mbed_tools_ci/utils/git_helpers.py

```diff
@@ -2,21 +2,33 @@
 import logging
 import os
 import re
-from git import Repo, Actor
+import shutil
+from pathlib import Path
 from typing import Optional, List, Union, Any, Tuple
 
+from git import Repo, Actor
+from packaging import version
+
 from .configuration import configuration, ConfigurationVariable
+from .filesystem_helpers import TemporaryDirectory
 
 logger = logging.getLogger(__name__)
 
 
 class GitWrapper:
     """Wrapper class to provide convenient methods for performing git actions."""
 
-    def __init__(self) -> None:
-        """Creates an instance of GitWrapper."""
-        self.repo = Repo(
-            configuration.get_value(ConfigurationVariable.PROJECT_ROOT))
+    def __init__(self, path: Optional[str] = None,
+                 repo: Optional[Repo] = None) -> None:
+        """Creates an instance of GitWrapper.
+
+        Args:
+            path: path to repository. If None, project root is considered.
+            repo: GitPython repository. if None, the repository is populated.
+        """
+        self._root_path = Path(path if path else configuration.get_value(
```

Comment:
I think this should take similar treatment as `GitTempClone` and `ProjectTempClone` had. I.e.: separate construction of a known object to a separate class. Less braching in `__init__` is always a win.